src/conf.rs

```diff
@@ -297,20 +298,49 @@ pub fn create_default_config(path: &PathBuf) -> Result<String, io::Error> {
 
 mod tests {
     // We'll use this git repo for testing
+    use std::env;
     use std::fs::{File,OpenOptions,remove_file};
-    use std::io::{Write,Read};
+    use std::io::{Read};
     use std::path::{Path,PathBuf};
+    use std::process::{Command,Stdio};
     use conf::{get_configuration,create_default_config,DEFAULT_CONF,Conf};
     use yaml_rust::{YamlLoader};
     use backend::Backend;
     use models::DisplayMaster;
-    use git2::Repository;
+    use git2::{Repository};
+    use tempdir::TempDir;
+
+    macro_rules! init_git {
+        () => {
+            let dir = TempDir::new("p-g-p").unwrap();
+            env::set_current_dir(&dir).unwrap();
+            // we could use git2 to create a repo with a commit, but it's soooo complicated
```

Comment:
is it, really?

Reply:
it was, back then, I should totally revisit this and finally fix it.